core/src/main/scala/org/apache/spark/streaming/SnappyStreamingContext.scala

```diff
@@ -56,7 +59,28 @@ class SnappyStreamingContext protected[spark](
         "both SparkContext and checkpoint as null")
   }
 
-  val snappySession = new SnappySession(sc)
+  val snappySession = reuseSnappySession.getOrElse(new SnappySession(sc))
+  currentSnappySession.foreach(csn => {
+    var user = csn.conf.get(Attribute.USERNAME_ATTR, "")
+
+    val isSmartConnProp = if (user.isEmpty) {
+      // In smart connector, property name is different.
+      user = csn.conf.get(Constant.SPARK_STORE_PREFIX + Attribute.USERNAME_ATTR, "")
+      true
+    } else {
+      false
+    }
+    if (!user.isEmpty) {
+      val password = csn.conf.get(Attribute.PASSWORD_ATTR, "")
```

Comment:
Why not checking isSmartConnProp for password too?

Reply:
Thanks Amogh for pointing that out. I was under the impression only user name is having different attrib. but looks like password to has different key